src/InOneWeekend/camera.h

```diff
@@ -60,6 +56,7 @@ class camera {
     }
 
   private:
+    int image_height;
```

Comment:
it's kind of weird having image_width be public and image_height be private?

Consider having image_height be public and just solve for any of the image_height, image_width, aspect ratio that are not explicitly defined? ...

No. That doesn't really work. If they're all public it's hard to know which were explicitly set unless we create constructors. But that's not a fantastic idea if we let main set aspect_ratio.

What was the decision behind having aspect ratio and width be set by main? I don't remember.

We shouldn't do getters/setters (we decided no a while ago).

WDYT?

Reply:
Three variables, two unknowns. The main idea is that while developing, people will typically alter parameters to adjust speed of rendering, image dimensions being one aspect. Once the render aspect ratio is done, it's typically fixed. So they can just adjust a single intuitive variable (width) or two variables plus at least a mental calculation. The former is a nicer interface. Also, if you want to compare renders without regard to size, then you want a fixed aspect ratio.